Makefile

```diff
@@ -2,53 +2,48 @@
 CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
 
 .PHONY: all deploy build helm
+.ONESHELL: test
 all: help
 
-help:
-	@echo "make miniup: start minikube"
-	@echo "make minidown: stop minikube"
-	@echo "make minidelete: delete minikube"
-	@echo "make minidashboard: open minikube dashboard"
-	@echo "make build: build db-operator docker image"
-	@echo "make helm: install helm if not exist and install local chart using helm upgrade --install command"
-	@echo "make setup: build db-operator image, install helm"
-	@echo "make update: build db-operator image again and delete running pod"
-	@echo "make addexamples: kubectl create -f examples/"
-	@echo "make test: spin up mysql, postgres containers and run go unit test"
-	@echo "make microsetup: install microk8s locally and deploy db-operator (only for linux and mac)"
-	@echo "make manifests: generate custom resource definitions"
-	@echo "make generate: generate supporting code for custom resource types"
-
-build:
+help:   ## show this help
+	@echo 'usage: make [target] ...'
+	@echo ''
+	@echo 'targets:'
+	@egrep '^(.+)\:\ .*##\ (.+)' ${MAKEFILE_LIST} | sort | sed 's/:.*##/#/' | column -t -c 2 -s '#'
+
+build: ## build db-operator docker image
 	@docker build -t my-db-operator:local .
 	@docker save my-db-operator > my-image.tar
 	@docker build -t mock-googleapi:local mock/
 	@docker save mock-googleapi > mock-google-api.tar
 
-helm:
+helm: ## install helm if not exist and install local chart using helm upgrade --install command
 	@helm upgrade --install --create-namespace --namespace operator my-dboperator helm/db-operator -f helm/db-operator/values.yaml -f helm/db-operator/values-local.yaml
 
-helm-lint:
+helm-lint: ## lint helm manifests
 	@helm lint -f helm/db-operator/values.yaml -f helm/db-operator/ci/ci-1.yaml --strict ./helm/db-operator
 	@helm lint -f helm/db-instances/values.yaml --strict ./helm/db-instances
 
-addexamples:
+addexamples: ## add examples via kubectl create -f examples/
 	cd ./examples/; ls | while read line; do kubectl apply -f $$line; done
 
-setup: build helm
+setup: build helm ## build db-operator image, install helm
 
 deploy:
 	@kubectl delete pod -l app=db-operator -n operator &
 	watch -n0.2 -c 'kubectl logs -l app=db-operator --all-containers=true -n operator'
 
-update: build deploy
+update: build deploy ## build db-operator image again and delete running pod
 
-test:
-	@docker-compose up -d
-	@docker-compose restart sqladmin
-	@sleep 2
-	@go test -count=1 -tags tests ./... -v -cover
-	@docker-compose down
+test: ## spin up mysql, postgres containers and run go unit test
+	tearDown() {
```

Comment:
avoid docker-compose still run if the target failed. Think mainly an issue on local side.

Reply:
Does this work? Definitely an interesting construct to see in a makefile.

Reply:
yes, tested before fixed my unit test error.